rstream/client.py

```diff
@@ -147,7 +149,8 @@ async def send_frame(self, frame: schema.Frame) -> None:
         logger.debug("Sending frame: %s", frame)
         assert self._conn
         try:
-            await self._conn.write_frame(frame)
+            if self.is_connection_alive():
+                await self._conn.write_frame(frame)
         except socket.error:
```

Comment:
Here, you should add `BrokenPipeError` to handle the write error during the close, like:
```  File "/Users/gas/git/qweeze/rstream/rstream/client.py", line 223, in _listener
      frame = await self._conn.read_frame()
    File "/Users/gas/git/qweeze/rstream/rstream/connection.py", line 73, in read_frame
      return decode_frame(await self._read_frame_raw())
    File "/Users/gas/git/qweeze/rstream/rstream/connection.py", line 67, in _read_frame_raw
      return await self._read(length)
    File "/Users/gas/git/qweeze/rstream/rstream/connection.py", line 52, in _read
      self._buffer += await self._reader.read(CHUNK_SIZE)
    File "/usr/local/Cellar/python@3.10/3.10.11/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/streams.py", line 669, in read
      await self._wait_for_data('read')
    File "/usr/local/Cellar/python@3.10/3.10.11/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/streams.py", line 501, in _wait_for_data
      await self._waiter
    File "/usr/local/Cellar/python@3.10/3.10.11/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/selector_events.py", line 949, in _write_ready
      n = self._sock.send(self._buffer)
  BrokenPipeError: [Errno 32] Broken pipe
  
  During handling of the above exception, another exception occurred:
  
  Traceback (most recent call last):
    File "/Users/gas/git/qweeze/rstream/rstream/client.py", line 242, in _listener
      await result
  RuntimeError: Event loop is closed
```